tck/src/main/java/org/eclipse/microprofile/context/tck/ThreadContextTest.java

```diff
@@ -1426,4 +1412,122 @@ public void currentContextExecutorRunsWithContext() throws InterruptedException,
+                .build();
+        ThreadContext labelContext = contextManager.newThreadContextBuilder()
+                .propagated(Label.CONTEXT_NAME)
+                .unchanged()
+                .cleared(ThreadContext.ALL_REMAINING)
+                .build();
+        try {
+            // Set non-default values
+            Label.set("default-executor-service-context-test-label-A");
+            
+            CompletableFuture<String> noContextCF = new CompletableFuture<>();
+            CompletableFuture<String> noContextCFStage1 = noContextCF.thenApplyAsync(res -> {
+                Assert.assertEquals(Label.get(), "",
+                        "Context type should not be propagated to non-contextual action.");
+
+                return res;
+            });
+
+            CompletableFuture<String> contextCF = labelContext.withContextCapture(noContextCFStage1).thenApplyAsync(res -> {
+                Assert.assertEquals(Label.get(), "default-executor-service-context-test-label-A",
+                        "Context type should be propagated to contextual CompletableFuture.");
+                
+                Assert.assertTrue(Thread.currentThread() instanceof TckThread,
+                        "Current thread should have been created by default executor service");
+                return res;
+            });
+
+            CompletionStage<String> contextCS = labelContext.withContextCapture((CompletionStage<String>)noContextCFStage1).thenApplyAsync(res -> {
+                Assert.assertEquals(Label.get(), "default-executor-service-context-test-label-A",
+                        "Context type should be propagated to contextual CompletionStage.");
+                
+                Assert.assertTrue(Thread.currentThread() instanceof TckThread,
+                        "Current thread should have been created by default executor service");
+                return res;
+            });
+
+            noContextCF.complete("OK");
+            contextCF.get(MAX_WAIT_NS, TimeUnit.NANOSECONDS);
+            contextCS.toCompletableFuture().get(MAX_WAIT_NS, TimeUnit.NANOSECONDS);
+        }
+        finally {
+            // Restore original values
+            Label.set(null);
+            executorService.shutdownNow();
+        }
+    }
+
+    /**
+     * Verify that the lack of a default executor service bans contextualised CompletableFuture/CompletionStage
+     * from running async actions on them.
+     *
+     * @throws ExecutionException indicates test failure
+     * @throws InterruptedException indicates test failure
+     * @throws TimeoutException indicates test failure
+     */
+    @Test
+    public void withoutDefaultExecutorServiceContextCannotInvokeAsyncActions()
+            throws ExecutionException, InterruptedException, TimeoutException {
+        ContextManager contextManager = ContextManagerProvider
+                .instance()
+                .getContextManagerBuilder()
```

Comment:
Same as previous comment - getContextManagerBuilder is optional, so test needs to skip if not implemented.